docs/spec_draft.md

```diff
@@ -703,6 +704,76 @@ matrix, then the behavior is undefined.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.compare
+
+### Semantics
+
+Performs element-wise comparison of `lhs` and `rhs` tensors according to
+`comparison_direction` and `compare_type`, and produces a `result` tensor.
+
+The values of `comparison_direction` have the following semantics:
+  * `EQ`: `lhs` $=$ `rhs`.
+  * `NE`: `lhs` $\ne$ `rhs`.
+  * `GE`: `lhs` $\ge$ `rhs`.
+  * `GT`: `lhs` $\gt$ `rhs`.
+  * `LE`: `lhs` $\le$ `rhs`.
+  * `LT`: `lhs` $\lt$ `rhs`.
+
+The values of `compare_type` have the following semantics:
+  * `FLOAT`: If element-type is floating-type, ordered floating-point comparison
+             of `lhs` and `rhs` is performed for all values of
+             `comparison_direction` except for `NE` for which unordered
+             comparison is performed. If element-type is complex-type,
+             lexicographic comparison of (real, imag) pairs is performed.
+  * `TOTALORDER`: `totalOrder` floating-point comparison of `lhs` and `rhs` as
+                  specified in IEEE-754.
+  * `SIGNED`: Signed comparison of `lhs` and `rhs`.
+  * `UNSIGNED`: Unsigned comparison of `lhs` and `rhs`.
```

Comment:
Proposed rewording:

```
The values of `comparison_direction` and `compare_type` have the following 
semantics:

For integer and boolean element types:
  * `EQ`: `lhs` $=$ `rhs`.
  * `NE`: `lhs` $\ne$ `rhs`.
  * `GE`: `lhs` $\ge$ `rhs`.
  * `GT`: `lhs` $\gt$ `rhs`.
  * `LE`: `lhs` $\le$ `rhs`.
  * `LT`: `lhs` $\lt$ `rhs`.

For floating-point element types and `compare_type = FLOAT`, the op implements
the following IEEE-754 operations:
  * `EQ`: `compareQuietEqual`.
  * `NE`: `compareQuietNotEqual`.
  * `GE`: `compareQuietGreaterEqual`.
  * `GT`: `compareQuietGreater`.
  * `LE`: `compareQuietLessEqual`.
  * `LT`: `compareQuietLess`.

For floating-point element types and `compare_type = TOTALORDER`, the op
uses the combination of `totalOrder` and `compareQuietEqual` operations from 
IEEE-754.

For complex element types, lexicographic comparison of `(real, imag)` pairs is 
performed using the provided `comparison_direction` and `compare_type`.
```

Feedback:
  1. Need to go into detail about what `comparison_direction` values mean for floats.
  2. Sounds like we can safely ignore `compare_type` for non-floats.
  3. I think it's better to structure semantics discussion by element type rather than by attribute values.
  4. Not exactly sure how `TOTALORDER` is supposed to work for `EQ`, `NE`, `GE` and `LE`. `totalOrder` from IEEE-754 doesn't have the "or equal" flavor. I made a guess of how it could work, would be good to verify it.